helpers.py

```diff
@@ -33,3 +33,15 @@ def setup_logger():
     console.setFormatter(logging.Formatter(log_format, datefmt=log_datefmt))
     logging.getLogger('').addHandler(console)
     logging.getLogger('urllib3').setLevel(logging.WARNING)
+
+
+def parse_node_type_from_proposal(proposal):
+    service_definition = proposal.get('service_definition', None)
+    if not service_definition:
+        return None
+
+    location = service_definition.get('location', None)
```

Comment:
`{service_definition: 'test'}` will throw exception here, we should probably handle this and return `None` if we declare that we return `None` in case we are unable to parse value.